src/sdam/state/mod.rs

```diff
@@ -107,6 +113,7 @@ impl Topology {
         let mut topology_state = TopologyState {
             description,
             servers: Default::default(),
+            http_client: Default::default(),
```

Comment:
I [checked with the people who develop hyper](https://github.com/hyperium/hyper/issues/2228) (the HTTP library that reqwest wraps) to ensure that creating a client and then never using it doesn't incur any runtime penalties, which luckily is the case. This means that instead of passing around an Option and then populating it lazily (which would require a lock, since we can't otherwise modify an Option that's cloned across multiple threads), we can eagerly create the client, and if the client doesn't use AWS auth, we don't pay any costs (e.g. extra threads spun up) other than the memory to allocate it.